### PR TITLE
Feature/new component error view

### PR DESCRIPTION
**Pull request by cfpava**
new view component into emerlad textfield.

image

### CODE REVIEW

File: EmeraldIOS/Components/TextfieldView/EmeraldInputViewValidableType.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    func validateEmeraldFields(with fields: [EmeraldInputViewValidableType]) -> Bool
+}
+
+extension EmeraldInputViewValidableType {
+    public func validateEmeraldFields(with fields: [EmeraldInputViewValidableType]) -> Bool {
+        return fields.map({ $0.validateAndHandle() }).filter({ $0 == false }).isEmpty
+    }
+    
+    public func validateAndHandle() -> Bool {
+        return true
```
Comment (Sgiraldo10) [inline EmeraldIOS/Components/TextfieldView/EmeraldInputViewValidableType.swift]: Should this always return `true`?

File: ProofOfConcept/ProofOfConcept/ViewController.swift [hunk 2]
```diff
@@ -41,6 +41,9 @@ class ViewController: UIViewController, EmeraldValidableType {
     @IBOutlet weak var emeraldButtonByStory: EmeraldButton!
     @IBOutlet weak var emeraldSelectorByStory: EmeraldSelectorField!
 
+//    @IBOutlet weak var emeraldRexgexTextFieldView: RegexTextFieldView!
```
Comment (Sgiraldo10) [inline ProofOfConcept/ProofOfConcept/ViewController.swift]: Remove this.

File: EmeraldIOS/Components/TextfieldView/EmeraldInputViewValidableType.swift (hunk 1 above)
Comment (cfpava) [reply EmeraldIOS/Components/TextfieldView/EmeraldInputViewValidableType.swift]: Done, remove method.

File: ProofOfConcept/ProofOfConcept/ViewController.swift (hunk 2 above)
Comment (cfpava) [reply ProofOfConcept/ProofOfConcept/ViewController.swift]: Done.

File: EmeraldIOS/Components/TextfieldView/EmeraldTextDependantFieldView.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    public override var textField: EmeraldTextField {
+        get {
+            return emeraldTextDependantField
+        }
+        set {
+            emeraldTextDependantField = newValue as! EmeraldTextDependantField
+        }
+    }
+
```
Comment (yhernandezs) [inline EmeraldIOS/Components/TextfieldView/EmeraldTextDependantFieldView.swift]: remove break line

**Pull request merged**